bindgen/ir/IR.cpp

```diff
@@ -230,18 +230,22 @@ void IR::replaceTypeInTypeDefs(std::shared_ptr<Type> oldType,
 
 template <typename T>
 bool IR::isTypeUsed(const std::vector<T> &declarations,
-                    std::shared_ptr<Type> type) {
-    for (const auto decl : declarations) {
-        if (decl->usesType(type)) {
+                    std::shared_ptr<Type> type, bool stopOnTypeDefs) {
+    for (const auto &decl : declarations) {
+        if (decl->usesType(type, stopOnTypeDefs)) {
             return true;
         }
     }
     return false;
 }
 
 bool IR::typeIsUsedOnlyInTypeDefs(std::shared_ptr<Type> type) {
-    return !(isTypeUsed(functions, type) || isTypeUsed(structs, type) ||
-             isTypeUsed(unions, type));
+    /* varDefines are not checked here because they are simply
+     * aliases for variables.*/
+    return !(
+        isTypeUsed(functions, type, true) || isTypeUsed(structs, type, true) ||
+        isTypeUsed(unions, type, true) || isTypeUsed(variables, type, true) ||
+        isTypeUsed(literalDefines, type, true));
```

Comment:
It looks like `stopOnTypeDefs` is always `true`. So I am wondering why we need that parameter?

Reply:
Because I expect that default behavior of `usesType` is to go through whole chain of types.
If we remove this parameter then `usesType` will always return `false` when we check whether `fun` uses `CInt`, this feels wrong to me.
```c
typedef int size;
size fun();
```
So it just makes sure that we are aware of the behavior of `usesType`.

Reply:
Is it ok?

Reply:
I added usage of `false` to a new branch, so I'll merge this one